cristin-organization/src/test/java/no/unit/nva/cristin/organization/QueryCristinOrganizationHandlerTest.java

```diff
@@ -0,0 +1,130 @@
+import nva.commons.core.Environment;
+import nva.commons.core.JsonUtils;
+import nva.commons.core.paths.UriWrapper;
+import org.apache.http.entity.ContentType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.zalando.problem.Problem;
+
+import javax.ws.rs.core.HttpHeaders;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static no.unit.nva.cristin.projects.Constants.BASE_PATH;
+import static no.unit.nva.cristin.projects.Constants.DOMAIN_NAME;
+import static no.unit.nva.cristin.projects.Constants.HTTPS;
+import static no.unit.nva.cristin.projects.Constants.ORGANIZATION_PATH;
+import static no.unit.nva.cristin.projects.ErrorMessages.ERROR_MESSAGE_QUERY_MISSING_OR_HAS_ILLEGAL_CHARACTERS;
+import static no.unit.nva.testutils.RandomDataGenerator.randomString;
+import static nva.commons.apigateway.ApiGatewayHandler.ALLOWED_ORIGIN_ENV;
+import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class QueryCristinOrganizationHandlerTest {
+
+    public static final ObjectMapper restApiMapper = JsonUtils.dtoObjectMapper;
+    QueryCristinOrganizationHandler queryCristinOrganizationHandler;
+    private CristinApiClient cristinApiClient;
+    private Environment environment;
+    private ByteArrayOutputStream output;
+    private Context context;
+
+
+
+    @BeforeEach
+    public void setUp() throws HttpClientFailureException, InterruptedException, NonExistingUnitError {
+
+        environment = mock(Environment.class);
+        when(environment.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn("*");
+        context = mock(Context.class);
+        cristinApiClient = mock(CristinApiClient.class);
+        when(cristinApiClient.queryInstitutions(any())).thenReturn(emptySearchResponse());
+
+        output = new ByteArrayOutputStream();
+        queryCristinOrganizationHandler = new QueryCristinOrganizationHandler(cristinApiClient, environment);
+    }
+
+    private SearchResponse<Organization> emptySearchResponse() {
+        return  new SearchResponse<Organization>(randomId()).withHits(Collections.emptyList());
+    }
+
+
+
+    @Test
+    public void returnsBadRequestResponseOnMissingQueryParam() throws IOException {
+        InputStream inputStream = generateHandlerRequestWithMissingQueryParameter();
+        queryCristinOrganizationHandler.handleRequest(inputStream, output, context);
+        GatewayResponse<Problem> gatewayResponse = parseFailureResponse();
```

Comment:
Since we're using Problem, create a method that tests Problem expectations and re-use it